packages/nextjs/src/config/wrappers/wrapperUtils.ts

```diff
@@ -1,25 +1,46 @@
+import { getActiveTransaction } from '@sentry/tracing';
+
 import { DataFetchingFunction } from './types';
 
 /**
- * Pass-through wrapper for the original function, used as a first step in eventually wrapping the data-fetching
- * functions with code for tracing.
+ * Create a span to track the wrapped function and update transaction name with parameterized route.
  *
  * @template T Types for `getInitialProps`, `getStaticProps`, and `getServerSideProps`
  * @param origFunction The user's exported `getInitialProps`, `getStaticProps`, or `getServerSideProps` function
  * @param context The context object passed by nextjs to the function
  * @returns The result of calling the user's function
  */
-export async function callOriginal<T extends DataFetchingFunction>(
-  origFunction: T['fn'],
-  context: T['context'],
-): Promise<T['result']> {
-  let props;
-
-  // TODO: Can't figure out how to tell TS that the types are correlated - that a `GSPropsFunction` will only get passed
-  // `GSPropsContext` and never, say, `GSSPContext`. That's what wrapping everything in objects and using the generic
-  // and pulling the types from the generic rather than specifying them directly was supposed to do, but... no luck.
-  // eslint-disable-next-line prefer-const, @typescript-eslint/no-explicit-any
-  props = await (origFunction as any)(context);
-
-  return props;
+export async function wrapperCore<T extends DataFetchingFunction>(options: {
+  origFunction: T['fn'];
+  context: T['context'];
+  route: string;
+  op: string;
+}): Promise<T['result']> {
+  const { origFunction, context, route, op } = options;
+
+  const transaction = getActiveTransaction();
```

Comment:
We probably also wanna add a `hasTracingEnabled()` check here.

Reply:
How could there be an active transaction if it's not, though?

Reply:
You're right! Sorry.